File: apps/server/src/api/health.ts
// Module: Exposes API health endpoints for service monitoring.
import { Elysia } from "elysia"

import type { Settings } from "../core/config"
import { HealthResponseSchema } from "../schemas/health"

export function createHealthRoutes(settings: Settings) {
  return new Elysia({ name: "health-routes" }).get(
    "/health",
    () => ({
      status: "ok" as const,
      service: settings.appName,
      version: settings.apiVersion,
      environment: settings.environment,
    }),
    {
      response: HealthResponseSchema,
    }
  )
}

File: apps/server/app/api/routes/health.py
# Module: Exposes API health endpoints for service monitoring.
from fastapi import APIRouter

from app.core.config import get_settings
from app.schemas.health import HealthResponse

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
def health_check() -> HealthResponse:
    settings = get_settings()

    return HealthResponse(
        status="ok",
        service=settings.app_name,
        version=settings.api_version,
        environment=settings.environment,
    )
